paasta_tools/cli/cmds/status.py

```diff
@@ -294,11 +311,20 @@ def paasta_status_on_api_endpoint(
     if status.git_sha != "":
         output.append("    Git sha:    %s (desired)" % status.git_sha)
 
-    instance_type = find_instance_type(status)
-
-    if system_paasta_config.get_enable_custom_cassandra_status_writer():
-        if status.get("cassandracluster") is not None:
-            instance_type = "cassandracluster"
+    if kubernetes:
+        instance_type = KUBERNETES_INSTANCE_TYPE
+        if status.get(instance_type) is None:
+            output.append(
+                PaastaColors.red(f"Status contains no data for {instance_type}.")
+            )
+            return 1
+    else:
+        instance_type = find_instance_type(status)
+        for instance_type_cr in INSTANCE_TYPES_K8S_CR:
+            if status.get(instance_type_cr) is not None:
+                # By default, the instance type is the CR instance type:
+                instance_type = instance_type_cr
+                break
```

Comment:
how come we can't just use the instance type returned from `find_instance_type`?